server/container/core/src/main/java/org/apache/james/server/core/MailImpl.java

```diff
@@ -100,13 +94,35 @@
      * @throws MessagingException when the message is not clonable
      */
     public static MailImpl duplicate(Mail mail) throws MessagingException {
-        return new MailImpl(mail, deriveNewName(mail.getName()));
+        return MailImpl.builder()
+            .name(deriveNewName(mail.getName()))
+            .sender(mail.getMaybeSender())
+            .addRecipients(mail.getRecipients())
+            .mimeMessage(new MimeMessageCopyOnWriteProxy(mail.getMessage()))
+            .remoteHost(mail.getRemoteHost())
+            .remoteAddr(mail.getRemoteAddr())
+            .lastUpdated(mail.getLastUpdated())
+            .errorMessage(mail.getErrorMessage())
+            .addAttributes(duplicateAttributes(mail))
+            .build();
+    }
+
+    private static ImmutableList<Attribute> duplicateAttributes(Mail mail) {
+        try {
+            return mail.attributes().map(Attribute::duplicate).collect(Guavate.toImmutableList());
+        } catch (IllegalStateException e) {
+            LOGGER.error("Error while cloning Mail attributes", e);
```

Comment:
To be honest, I decided not to chance existing behavior and this code has just been move, that's why it's an error.
I agree it could be a warning but should we handle it now?

Reply:
That's fine for me.